src/dfx-core/src/config/model/local_server_descriptor.rs

```diff
@@ -146,6 +146,16 @@ impl LocalServerDescriptor {
         self.replica_configuration_dir().join("replica-pid")
     }
 
+    /// This file contains the listening port of the pocket-ic process
+    pub fn pocketic_port_path(&self) -> PathBuf {
+        self.data_directory.join("pocket-ic.port")
+    }
+
+    /// Returns whether the local server is PocketIC (as opposed to the replica)
+    pub fn is_pocketic(&self) -> bool {
+        self.pocketic_port_path().exists()
```

Comment:
This probably won't work well with the call to `empty_writable_path` in start.rs, which unconditionally creates an empty file.  

https://github.com/dfinity/sdk/pull/3761/files#diff-04d9df0d52b6d87bc92f994bdc87cfe37982f42e3db092347d0c5b688380f8adR246